hash_cache_test.go

```diff
@@ -111,18 +111,23 @@ func BenchmarkHashWithCache(b *testing.B) {
 func TestBlockCache_maxSize(t *testing.T) {
 	maxCacheSize := int64(10000)
 	cache := newHashCache(maxCacheSize)
-	for i := uint64(0); i < uint64(maxCacheSize+1025); i++ {
+	accessSize := int64(500)
+	for i := uint64(0); i < uint64(maxCacheSize+accessSize); i++ {
 		rootNum := make([]byte, 8)
 		binary.LittleEndian.PutUint64(rootNum, i)
 		if err := cache.AddRoot(toBytes32(rootNum[:4]), []byte{1}); err != nil {
 			t.Fatal(err)
 		}
 	}
-	log.Printf(
-		"hash cache key size is %d, itemcount is %d",
-		maxCacheSize,
-		cache.hashCache.ItemCount(),
-	)
+	for int64(cache.hashCache.ItemCount()) > maxCacheSize && cache.hashCache.ItemCount() <= int(maxCacheSize+accessSize) {
+		log.Printf(
+			"hash cache key size is %d, itemcount is %d",
+			maxCacheSize,
+			cache.hashCache.ItemCount(),
+		)
+		time.Sleep(1)
```

Comment:
Can we avoid using sleep? I think @rauljordan has addressed similar concerns as well

Reply:
+1, sleep makes a test brittle

Reply:
this time i am waiting till it gets pruned or test timeout . it is not brittle as the constant sleep before. i would like to send a message on a channel while gc() function is being called but it is part of the ccache package and not owned by me. any better suggestion ?